src/main/java/com/google/firebase/auth/hash/Argon2.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.auth.hash;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.BaseEncoding;
+import com.google.firebase.auth.UserImportHash;
+import java.util.Map;
+
+/**
+ * Represents the Argon2 password hashing algorithm. Can be used as an instance of {@link
+ * com.google.firebase.auth.UserImportHash} when importing users.
+ */
+public final class Argon2 extends UserImportHash {
+
+  private final int hashLengthBytes;
+  private final Argon2HashType hashType;
+  private final int parallelism;
+  private final int iterations;
+  private final int memoryCostKib;
+  private final Argon2Version version;
+  private final String associatedData;
+
+  private Argon2(Builder builder) {
+    super("ARGON2");
+    checkArgument(builder.hashLengthBytes >= 4 && builder.hashLengthBytes <= 1024,
+        "hashLengthBytes is required for Argon2 and must be between 4 and 1024");
+    checkArgument(builder.hashType != null,
+        "A hashType is required for Argon2");
+    checkArgument(builder.parallelism >= 1 && builder.parallelism <= 16,
```

Comment:
nit: consider making the limit values as constants.

Reply:
Done